python/cudf/cudf/tests/test_avro_reader_fastavro_integration.py

```diff
@@ -0,0 +1,180 @@
+            "namespace": namespace,
+            "fields": [{"name": "prop", "type": avro_type}],
+        }
+    )
+
+    actual = cudf_from_avro_util(schema, [])
+
+    expected = cudf.DataFrame(
+        {"prop": cudf.Series(None, None, expected_dtype)}
+    )
+
+    assert_eq(expected, actual)
+
+
+@pytest.mark.parametrize("avro_type, expected_dtype", avro_type_params)
+@pytest.mark.parametrize("namespace", [None, "root_ns"])
+@pytest.mark.parametrize("nullable", [True, False])
+def test_can_detect_dtype_from_avro_type_nested(
+    avro_type, expected_dtype, namespace, nullable
+):
+    avro_type = avro_type if not nullable else ["null", avro_type]
+
+    schema_leaf = {
+        "name": "leaf",
+        "type": "record",
+        "fields": [{"name": "prop3", "type": avro_type}],
+    }
+
+    schema_child = {
+        "name": "child",
+        "type": "record",
+        "fields": [{"name": "prop2", "type": schema_leaf}],
+    }
+
+    schema_root = {
+        "name": "root",
+        "type": "record",
+        "namespace": namespace,
+        "fields": [{"name": "prop1", "type": schema_child}],
+    }
+
+    actual = cudf_from_avro_util(schema_root, [])
+
+    col_name = "{ns}child.{ns}leaf.prop3".format(
+        ns="" if namespace is None else namespace + "."
+    )
+
+    expected = cudf.DataFrame(
+        {col_name: cudf.Series(None, None, expected_dtype)}
+    )
+
+    assert_eq(expected, actual)
+
+
+@pytest.mark.parametrize(
+    "avro_type, cudf_type, avro_val, cudf_val",
+    [
+        ("boolean", "bool", True, True),
+        ("boolean", "bool", False, False),
+        ("int", "int32", 1234, 1234),
+        ("long", "int64", 1234, 1234),
+        ("float", "float32", 12.34, 12.34),
+        ("double", "float64", 12.34, 12.34),
+        ("string", "str", "heyϴ", "heyϴ"),
+        # ("bytes", "str", "heyϴ", "heyϴ"),
+    ],
+)
+def test_can_parse_single_value(avro_type, cudf_type, avro_val, cudf_val):
+
+    schema_root = {
+        "name": "root",
+        "type": "record",
+        "fields": [{"name": "prop", "type": ["null", avro_type]}],
+    }
+
+    records = [
+        {"prop": avro_val},
+    ]
+
+    actual = cudf_from_avro_util(schema_root, records)
+
+    expected = cudf.DataFrame(
+        {"prop": cudf.Series(data=[cudf_val], dtype=cudf_type)}
+    )
+
+    assert_eq(expected, actual)
+
+
+@pytest.mark.parametrize("avro_type, cudf_type", avro_type_params)
+def test_can_parse_single_null(avro_type, cudf_type):
+
+    schema_root = {
+        "name": "root",
+        "type": "record",
+        "fields": [{"name": "prop", "type": ["null", avro_type]}],
+    }
+
+    records = [{"prop": None}]
+
+    actual = cudf_from_avro_util(schema_root, records)
+
+    expected = cudf.DataFrame(
+        {"prop": cudf.Series(data=[None], dtype=cudf_type)}
+    )
+
+    assert_eq(expected, actual)
+
+
+@pytest.mark.xfail(
+    reason="cudf avro reader is unable to parse zero-field metadata."
+)
+def test_can_parse_empty_dataframe():
```

Comment:
[From an offline discussion]
We need three separate tests for "empty dataframe":

1. no schema, no data - should work;
2. empty schema, no data - newly uncovered bug;
3. valid schema, no data - should work;